File: java/utility/PrepareBody.java
package voxgig.bluefindecryptxp2pesdk.utility;

import voxgig.bluefindecryptxp2pesdk.core.Context;

final class PrepareBody {

  private PrepareBody() {}

  static Object prepareBody(Context ctx) {
    if ("data".equals(ctx.op.input)) {
      return ctx.utility.transformRequest.apply(ctx);
    }

    return null;
  }
}

File: csharp/utility/PrepareBody.cs
// BluefinDecryptxP2pe SDK utility: prepareBody.

namespace BluefinDecryptxP2peSdk.Util;

public static partial class SdkUtility
{
    internal static object? PrepareBodyUtil(Context ctx)
    {
        var op = ctx.Op!;

        if (op.Input == "data")
        {
            var body = ctx.Utility!.TransformRequest(ctx);
            return body;
        }

        return null;
    }
}
